modules/tracking/src/roiSelector.cpp

```diff
@@ -163,7 +163,7 @@ namespace cv {
     // while key is not ESC (27)
     while(key!=27){
       temp=select(windowName, img, true, fromCenter);
-      if(temp.width>0 && temp.height>0)
+      if(key!=27 && temp.width>0 && temp.height>0)
```

Comment:
Thanks!  I just pushed a cleaner version with your suggestion.  You're right that some people would never be caught by this, but when I handed the controls over to a novice, they were consistently confused about getting an extra rect if they had selected something *and then* hit escape.